utils/jwtUtil.js

```diff
@@ -56,25 +97,49 @@ function verifyToken(req, cb) {
   }
 
   if (token) {
-    jwt.verify(token, env.tokenSecret, {}, (err, decodedData) => {
+    jwt.verify(token, secret, {}, (err, decodedData) => {
       if (err) {
-        handleInvalidToken(cb);
-      } else {
-        User.findOne({ where: { name: decodedData.username } })
+        return handleInvalidToken(cb);
+      } else { // eslint-disable-line no-else-return
+        return User.findOne({ where: { name: decodedData.username } })
         .then((user) => {
-          // set user in request
           if (user) {
-            req.user = user;
-            return cb();
-          }
+            req.user = user; // set user in request
 
-          return handleInvalidToken(cb);
+            /*
+             * No need to check the token record if this is the default UI
+             * token.
+             */
+            if (decodedData.username === decodedData.tokenname) {
```

Comment:
After talking all of that over with you the other day, I convinced myself that I couldn't make that change without invalidating all the existing tokens, but I don't think that's true. In any case, this logic still needs to be there to handle the existing tokens, even if we start encoding new tokens with the extra field, so I think we should leave this as is, and create a new story to encode that new field into new tokens. What do you think, @pallavi2209?